js/app.js

```diff
@@ -43,3 +48,42 @@ document.getElementById("starting-items").textContent = getRandomWithSeed(
   .map(items => `C${items.id} ${items.name}`)
   .join(", ");
 document.getElementById("end-room").textContent = endingRoom;
+document.getElementById("generate-seed").addEventListener("click", function() {
+  newUrl = `?seed=${generateANewSeed()}`;
+  redirect(newUrl);
+});
+document.getElementById("share").addEventListener("click", function() {
+  copyToClipboard();
+});
+
+function generateANewSeed() {
+  let r = Math.random()
+    .toString(36)
+    .substring(7);
+  let rand = Math.floor(999 * Math.random());
+  return `${r}${rand}`;
+}
+
+function redirect(redirectUrl) {
+  window.location.replace(redirectUrl);
+}
+
+function copyToClipboard() {
+  // create new element to append url to so we can copy the text to clipboard
+  let newElement = document.createElement("input"),
```

Comment:
Multi-line assignments like this aren't really advised very often, as they create messy git diffs in the event of adding or removing assignments;

Multi-line assignments when adding a variable;
```Diff
 let value = 0,
-    otherValue = 3;
+    otherValue = 3,
+    yetAnotherValue = 4;
```

Multiple separate assignments;

```Diff
 let value = 0;
 let otherValue = 3;
+let yetAnotherValue = 4;
```

Reply:
I think your idea is better, I don't see a problem in terms of the code but definitely makes git diffs much easier to read. Cheers!

Reply:
Yeah I agree there's nothing _objectively_ wrong with the code, but as you said it's worth the change just to make diffs easier to read in the future.